src/evict/evict_page.c

```diff
@@ -248,7 +248,7 @@ __evict_page_dirty_update(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 	switch (mod->rec_result) {
 	case WT_PM_REC_EMPTY:				/* Page is empty */
 		/* Discard the parent's address. */
-		__wt_ref_free_addr(session, ref);
+		__wt_ref_addr_free(session, ref);
```

Comment:
Thanks, @keithbostic -- that's what I thought (that `ref->addr` should always be `NULL`), otherwise we could be leaking blocks there (which we clearly aren't, or verify would find them).

I've made the change you suggested: assert `ref->addr == NULL` at the top, remove these calls.  Nothing has fallen over yet.